plugins/com.google.cloud.tools.eclipse.appengine.whitelist/src/com/google/cloud/tools/eclipse/appengine/whitelist/AppEngineJreWhitelist.java

```diff
@@ -1901,11 +1911,55 @@
               ));
   
   /**
-   * @param className fully package qualified className
-   * @return true if this class is allowed in Java 7 on App Engine, false otherwise
+   * @param className fully package qualified class name
+   * @return true if this class is allowed in Java 7 on App Engine Standard, false otherwise
    */
   public static boolean contains(String className) {
-    return WHITELIST.contains(className);
+    if (className.startsWith("javax.") ){
+      if (isBundledInJre(className)) {
+        return WHITELIST.contains(className);
+      }
+      return true;
+    }
+    else if (className.startsWith("java.") 
+        || className.startsWith("sun.util.") 
+        || className.startsWith("org.xml.sax.") 
+        || className.startsWith("org.w3c.dom.") 
+        || className.startsWith("org.omg.") 
+        || className.startsWith("org.ietf.jgss") 
+        || className.startsWith("com.sun.")) {
+      return WHITELIST.contains(className);
+    } else { // not a JRE class
+      return true;
+    }
+    
+  }
+
+  private static boolean isBundledInJre(String className) {
+    if (className.startsWith("javax.accessibility")
+        || className.startsWith("javax.activation")
```

Comment:
Some of these have trailing dots and others don't. They should probably all have trailing dots?